tools/PowerShell/Microsoft.WinGet.Client/src/Library/Write-LogEntry.ps1

```diff
@@ -0,0 +1,55 @@
+Function Write-LogEntry
+{
+    PARAM(
+        [Parameter(Position=0, Mandatory=$true)]  [string] $LogEntry,
+        [Parameter(Position=1, Mandatory=$false)] [int]    $Severity=1,
+        [Parameter(Position=2, Mandatory=$false)] [string] $FontColor="",
+        [Parameter(Position=3, Mandatory=$false)] [int]    $Indent = 0,
+        [Parameter(Position=4, Mandatory=$false)] [switch] $NoNewLine
+    )
+    BEGIN
+    {
+        if($FontColor -eq "") {
+            switch ($Severity) {
+                "1" {
+                    ## Informational Response
+                    $FontColor     = "White"
+                    $MessagePreFix = ""
+                }
+                "2" {
+                    ## Warning Response
+                    $FontColor = "Yellow"
+                    $MessagePreFix = "WARNING:  "
+                }
+                "3" {
+                    ## Error Response
+                    $FontColor = "Red"
+                    $MessagePreFix = "ERROR:    "
+                }
+            }
+        }
+        ## Combines the logging message and the message type as a prefix
+        $LogEntry = $MessagePreFix + $LogEntry
+
+        ## Indents the message when viewed on the screen.
+        $LogIndent = "  "
+        while ($Indent -gt 0) {
+            $LogEntry = $LogIndent + $LogEntry
+            $Indent -= 1
+        }
```

Comment:
```suggestion
        $LogEntry = $LogEntry.PadLeft($LogEntry.Length + (2 * $Indent) )
```
I think this can be done without the while loop by calculating how long the string should be with the added indentation